contracts/tips/src/error.rs

```diff
@@ -0,0 +1,54 @@
+use cosmwasm_std::{StdError, Uint128};
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum ContractError {
+    #[error("{0}")]
+    Std(#[from] StdError),
+
+    #[error("unauthorized")]
+    Unauthorized {},
+
+    #[error("invalid tips history size, value: {value} max allowed: {max}")]
+    InvalidTipsHistorySize { value: u32, max: u32 },
+
+    #[error("invalid subspace id")]
+    InvalidSubspaceId {},
+
+    #[error("subspace with id {id} does not exist, {error}")]
+    SubspaceNotExist { id: u64, error: StdError },
+
+    #[error("invalid post id")]
+    InvalidPostId {},
+
+    #[error("post with id: {id} not found")]
+    PostNotFound { id: u64 },
+
+    #[error("to be able to tip you must have a profile")]
+    ProfileRequired {},
+
+    #[error("you can't tip yourself")]
+    SenderEqReceiver {},
+
+    #[error("provided a fee coin with value = 0, denom: {denom}")]
+    ZeroFeeCoin { denom: String },
+
+    #[error("empty fixed fee")]
+    EmptyFixedFee {},
+
+    #[error("invalid percentage fee")]
+    InvalidPercentageFee {},
+
+    #[error("insufficient {denom}, requested: {requested} provided: {provided}")]
+    InsufficientAmount {
+        denom: String,
+        requested: Uint128,
+        provided: Uint128,
+    },
```

Comment:
```suggestion
    #[error("insufficient fees, requested: {requested} provided: {provided}")]
    InsufficientAmount {
        requested: String,
        provided: String,
    },
```
I think it would be more clear to show all the tokens required, say, `insufficient fees, requested: 100uatom,100udsm provided: 100udsm`.